sable_network/src/network/event/details.rs

```diff
@@ -174,26 +174,28 @@ EventDetails => {
     }
 
     #[target_type(ChannelRegistrationId)]
-    struct ChannelRegistrationUpdate
-    {
+    struct ChannelRegistrationUpdate {
         pub data: Option<state::ChannelRegistration>,
     }
 
     #[target_type(ChannelAccessId)]
-    struct ChannelAccessUpdate
-    {
+    struct ChannelAccessUpdate {
         pub data: Option<state::ChannelAccess>,
     }
 
     #[target_type(ChannelRoleId)]
-    struct ChannelRoleUpdate
-    {
+    struct ChannelRoleUpdate {
         pub data: Option<state::ChannelRole>,
     }
 
     #[target_type(UserId)]
-    struct UserLogin
-    {
+    struct UserAway {
+        /// Empty iff the user is back from being away
+        pub reason: String,
```

Comment:
Is there a reason this isn't Option<String>?

Reply:
To follow the "Make Illegal States Unrepresentable" principle, as `Some("")` would be an invalid state.

Reply:
Fair, but I feel that in this case "" is sufficiently semantically different from any other string value that it should be different in the type system. Perhaps `Option<NonEmptyString>` or something similar?

Reply:
We'd be wasting a whole machine word of memory!

Reply:
Don e